### ISSUE TITLE
Copy hangs the clipboard in TextDiff

### ISSUE THREAD

**Issue by fabio-s-franco** [veler/DevToys#904]
### Current behavior

I have always had problems with copying text from DevToys and in more than one machine.

I don't know exactly what is happening on the background and I wish I had the time to go over this myself in the code, but what I can say is that similar to: #709 , I am also having problems.

The problem is not exactly the same, but I feel it might be related. And I also experienced the same in the past, where it behaved exactly like that.

What I noticed now though, is that it seems clipboard gets busy, whenever I copy something from DevToys. This time I noticed that after trying to copy something in DevToys, something really slow is pushing the text to the clipboard. So slow that if I try to copy something from anywhere else, it also doesn't work. I keep getting whatever was there before and it seems to happen more on text panels of the bottom side. After a little while, the clipboard seems to unchoke and it will past what was copied.

I think some people may miss this last detail because they keep trying to copy and it just hangs more and it never finishes, until they give up. Or until it "finally works".

In particular for my case, the problem happens very easily with Text Diff. Specially if it is a big diff (by big I mean > 200 lines).
It also looks like that this is more likely to happen when there is an accidental copy of the entire text (which happens quite frequently, when I select nothing, or when the mouse pointer goes out of bounds, it seems to copy everything).

Why one of the oldest windows native functions gets so slow in certain circumstances, I have no idea. But I believe something funky is going on that seems to make a copy operation feel as heavy as AAA games.

Never been a seamless experience with me.

Maybe this can problem can be spotted by somebody.

Thanks. Cheers,

### How to reproduce it (as minimally and precisely as possible)

1 - TextDiff -> Copy a reasonably large text into the editor in the left and something slightly different to compare into the editor to the right.

2 - Now, try to copy something from the diff result in the bottom. Try some different approaches as described (copy everything, then following that, select a small portion of text and copy again).
Can also be a good text and check the behavior when the mouse goes out of bounds just before hitting Ctrl+C

3 - Try to paste in notpad

4 - As I was validating my steps it happened again even with a small portion of text. DevToys hangs upon losing focus and responds after 15 seconds or so.

### Expected behavior

Whatever was copied last is immediately pasted into the editor

### Workaround

Unknown

### Affected platforms

Windows

### Affected DevToys kind

DevToys (app with GUI)

### DevToys Version

1.0.13.0

### Relevant Assets/Logs

Not about this particular case but I do have some clipboard errors in the past:

```
# - [05/25/2023 14:53:11]
Feature name: Failed to copy from code editor
Custom message: 
Exception message: OpenClipboard Failed (Exception from HRESULT: 0x800401D0)
Exception stack trace:
   at System.Runtime.InteropServices.McgMarshal.ThrowOnExternalCallFailed(Int32, RuntimeTypeHandle) + 0x21
   at __Interop.ComCallHelpers.Call(__ComObject, RuntimeTypeHandle, Int32) + 0xb8
   at __Interop.ForwardComStubs.Stub_17[TThis](__ComObject, Int32) + 0x24
   at Windows.ApplicationModel.DataTransfer.Clipboard.Flush() + 0x2c
   at DevToys.UI.Controls.CodeEditor.CopyButton_Click(Object, RoutedEventArgs) + 0xce
```

**Comment by veler** [comment]
Hi,
Thanks for this feedback. I didn't debug yet but I suspect the reason for this slowness is the following : 

The Editor in TextDiff us Monaco Editor, which is the same editor than Visual Studio Code, which runs in a web page. DevToys itself is a native app but since the Editor runs in a web page, there's a need of some interoperability to communicate between the web browser and the rest of the app. I suspect transmitting a large text between DevToys (native) and the web browser is what causes such a slowness.

The good news is that I already tested such a scenario in DevToys 2.0 (currently in development) and  it was quite satisfying. The main difference with DevToys 1.0 is that the whole app runs in a we browser, not just the Editor. Therefore there's no need of such interoperability, and in addition the Editor loads much faster (many users complain of a 1-2sdc delay to load the Editor in 1.0).

**Comment by fabio-s-franco** [comment]
@veler that's promising news. I am assuming VS Code itself uses the approach of 2.0, right? As I don't experience the issue there. Given this should be fixed in an upcoming 2.0 release, should this issue be kept open?

**Comment by veler** [comment]
>  I am assuming VS Code itself uses the approach of 2.0, right?

Yes! :D

> Given this should be fixed in an upcoming 2.0 release, should this issue be kept open?

Indeed, let's close this issue :)
